docs/synapse/userguides/ug015_storm_ref_pivot.rst

```diff
@@ -11,85 +15,339 @@ Where specific query examples are given, they are commonly provided in pairs usi
 
 pivot()
 -------
-Todo
+Returns a set of nodes that share a specified property of the same type / valu as the original set of nodes.
+
+Optional parameters:
+
+* **Return limit:** specify the maximum number of nodes returned by the ``pivot()`` query.
+
+  * ``limit=`` (operational syntax)
+  * ``^`` (macro syntax)
+
+**Operator syntax:**
+
+.. parsed-literal::
+
+  **pivot(** [ *<srcprop>* **,** ] *<dstprop>* [ **, limit=** *<limit>* ] **)**
+
+**Macro syntax:**
+
+.. parsed-literal::
+
+  [ *<srcprop>* **^** *<limit>* ] **->** *<dstprop>*
+
+**Examples:**
+
+* Pivot from a set of domains (``inet:fqdn`` nodes) in the working set to the DNS A records for those domains:
+  ::
+    pivot( inet:fqdn, inet:dns:a:fqdn )
+    
+    pivot( inet:dns:a:fqdn )
+    
+    inet:fqdn  ->  inet:dns:a:fqdn
+    
+    -> inet:dns:a:fqdn
+
+* Pivot from a set of domains in the working set to the DNS A records for those domains, but limit the results to 10 nodes:
+  ::
+    pivot( inet:fqdn, inet:dns:a:fqdn, limit=10 )
+    
+    pivot( inet:dns:a:fqdn, limit=10 )
+    
+    inet:fqdn^10  ->  inet:dns:a:fqdn
+
+* Pivot from a set of domains in the working set to the DNS A records for those domains, and from the IP addresses in the DNS A records to the associated set of IPv4 nodes (e.g., pivot from a set of domains to the set of IP addresses the domains resolved to).
+  ::
+    pivot( inet:fqdn, inet:dns:a:fqdn ) pivot( inet:dns:a:ipv4, inet:ipv4 )
+      
+    pivot( inet:dns:a:fqdn ) pivot( :ipv4, inet:ipv4 )
+    
+    inet:fqdn -> inet:dns:a:fqdn inet:dns:a:ipv4 
```

Comment:
I think this will lift `inet:fqdn` and pivot to `inet:dns:a:fqdn`, then lift w/ prop `inet:dns:a:ipv4', then pivot the whole working set to `inet:ipv4`. I think we can just remove `inet:dns:a:ipv4` from the query.

Reply:
I was definitely wrong about this, your example is correct. It is fairly confusing that the operator works this way in my opinion.